draft-duke-quic-load-balancers.md

```diff
@@ -505,6 +518,208 @@ reserved for config rotation {{config-rotation}}.
 The server then encrypts the second through seventeenth octets using the 128-bit
 AES-ECB cipher.
 
+# Retry Service {#retry-offload}
+
+When a server is under load, QUIC allows it to defer storage of connection state
+until the client proves it can receive packets at its advertised IP address.
+Through the use of a Retry packet, a token in subsequent client Initial
+packets, and the original_connection_id transport parameter, servers verify
+address ownership and clients verify that there is no "man in the middle"
+generating Retry packets.
+
+There are two different mechanisms to allow offload of DoS mitigation to a
+trusted network service. One requires no shared state; the server need only be
+configured to trust a retry service, though this imposes other operational
+constraints. The other requires shared key, but has no such constraints.
+
+## Common Requirements
+
+Regardless of mechanism, a retry service has an active mode, where it is
+generating Retry packets, and an inactive mode, where it is not, based on its
+assessment of server load and the likelihood an attack is underway. The choice
+of mode MAY be made on a per-packet basis, through a stochastic process or based
+on client address.
+
+A retry service MUST forward all packets for a QUIC version it does not
+understand. Note that if servers support versions the retry service does not,
+this may unacceptably increase loads on the servers. However, dropping these
+packets would introduce chokepoints to block deployment of new QUIC versions.
```

Comment:
It goes beyond this. The retry service must know which versions it can respond to with a Retry packet. Just because the retry service supports versions { 1, 2, 3 } doesn't mean the servers that is is providing the service for support them. If the server only supports { 2 }, they the retry service must only reply to { 2 }.

Reply:
I'm not sure that's workable. If new versions are incrementally deployed, there's no correct way to do this.

While the client receiving Retry and then Version Negotiation is not great, I'm not sure it's technically illegal. is it?

Reply:
We should confirm with the QUIC WG on what the required behavior is, but it seems really weird from a client perspective to get a Retry, reply to it, and then get a VN.